### PR TITLE
Enhance architecture documentation with additional links

### PR DESCRIPTION
**Pull request by andrewpollock**
* Link out to various GCP technologies to make it easier for the
  unfamiliar to school themselves
* Consistently use and place reference-style links for better readability

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -8,37 +8,47 @@ OSV runs on Google Cloud Platform, with the following main components:
...
+[models]:
+https://cloudplatform.googleblog.com/2015/08/Introduction-to-data-models-in-Cloud-Datastore.html
+[here]: https://github.com/google/osv/blob/master/osv/models.py
+
 ## OSS-Fuzz
 
-All of our data is currently sourced from
-[OSS-Fuzz](https://github.com/google/oss-fuzz), and we are working to extend
-this with other sources.
+All of our data is currently sourced from [OSS-Fuzz], and we are working to
```
Comment (oliverchang) [inline docs/architecture.md:21]: This is actually outdated. We source our data from many other places now. 

Can you instead just say this is an automated fuzzing service where we source a lot of our data and link to http://github.com/google/oss-fuzz ?

File: docs/architecture.md [hunk 2]
```diff
@@ -8,37 +8,47 @@ OSV runs on Google Cloud Platform, with the following main components:
 
 ## Cloud Datastore
 
-All vulnerability data is stored in Cloud Datastore, with the models defined
+All vulnerability data is stored in [Cloud Datastore], with the [models] defined
 [here].
 
+[Cloud Datastore]: https://cloud.google.com/datastore
+[models]:
+https://cloudplatform.googleblog.com/2015/08/Introduction-to-data-models-in-Cloud-Datastore.html
```
Comment (oliverchang) [inline docs/architecture.md:16]: This is very outdated and uses a different client library. Maybe link to https://googleapis.dev/python/python-ndb/latest/index.html#defining-entities-keys-and-properties instead?

File: docs/architecture.md (hunk 1 above)
Comment (andrewpollock) [reply docs/architecture.md:21]: How about I move this section to the top, change it to Sources and reword to say something along the line of the sources we currently draw from?

File: docs/architecture.md (hunk 2 above)
Comment (andrewpollock) [reply docs/architecture.md]: Yeah that's a good point, thanks. Updated in bc7de6a

File: docs/architecture.md (hunk 1 above)
Comment (andrewpollock) [reply docs/architecture.md]: I've done something along these lines in c3b4781, PTAL.

File: docs/architecture.md [hunk 3]
```diff
@@ -4,41 +4,50 @@
   <img src="images/architecture.png" width="800">
 </p>
 
+## Data source
+
+Our data is sourced from a variety of [sources], which we are looking to expand
+on over time.
+
+[sources]: https://github.com/google/osv.dev/tree/master/vulnfeeds
```
Comment (oliverchang) [inline docs/architecture.md:12]: Let's just link to https://github.com/google/osv.dev#current-data-sources

Comment (andrewpollock) [reply docs/architecture.md]: Addressed in be3378d

**Review by oliverchang — APPROVED** [review]

**Pull request merged**